src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -73,6 +73,11 @@ public static bool UseLazyResolution(MethodDesc method, string importModule, PIn
             }
         }
 
+        internal static bool IsRunningOnMono()
```

Comment:
if so, I can just remove this check

Reply:
We have added this check for `__Internal` to make libraries written for Mono work (https://github.com/dotnet/corert/pull/5247#issue-162806840). There is no default that will make everybody happy - it is why it needs to be a command line option.

Reply:
I jotted down the structure here: https://github.com/dotnet/corert/commit/2a4cac8e74515147572c7632587c09270918e588 but I ran out of time I allocated for myself to finish this.

@szhaomsft Can you apply this commit locally and help me finish it?

What we are going for is to allow specifying the list of pinvokes that should be bound directly as a command line option. One should be able to specify e.g. `--directpinvoke:"*" --directpinvoke:"api-ms-win*"` to specify these p/invokes should be hard bound.

The MSBuild targets file will make this flexible, so that once you finish this and we merge it, you'll be able to disable resolution of `__Internal` by adding an `Exclude` item for the `__Internal` entry in your CSPROJ file to disable the default rule.

Reply:
Submitted as #6980.